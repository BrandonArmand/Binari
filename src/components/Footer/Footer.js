import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { List, ListItem } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import styles from 'assets/jss/material-kit-react/components/footerStyle.js';

const useStyles = makeStyles(styles);

export default function Footer(props) {
  const classes = useStyles();
  const { whiteFont } = props;
  const footerClasses = classNames({
    [classes.footer]: true,
    [classes.footerWhiteFont]: whiteFont,
  });

  return (
    <footer className={footerClasses}>
      <div className={classes.container}>
        <div className={classes.left}>
          <List className={classes.list}>
            <ListItem className={classes.inlineBlock}>
              <a href="https://www.netlify.com">
                <img
                  src="https://www.vectorlogo.zone/logos/netlify/netlify-ar21.svg"
                  alt="Deploys by Netlify"
                />
              </a>
            </ListItem>
            <ListItem className={classes.inlineBlock}>
              <a
                href="http://brandonarmand.com"
                className={classes.block}
                target="_blank"
                rel="noopener noreferrer"
              >
                Brandon A. Welsh
              </a>
            </ListItem>
            <ListItem className={classes.inlineBlock}>
              <a
                href="https://github.com/BrandonArmand/Binari/blob/master/LICENSE.md"
                className={classes.block}
                target="_blank"
                rel="noopener noreferrer"
              >
                MIT License
              </a>
            </ListItem>
          </List>
        </div>
      </div>
    </footer>
  );
}

Footer.propTypes = {
  whiteFont: PropTypes.bool,
};
