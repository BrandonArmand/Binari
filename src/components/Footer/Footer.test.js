import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Footer from "./Footer";

describe("Footer", () => {
    it("renders snapshot", () => {
        const { container } = render(<Footer />);
        expect(container).toMatchSnapshot();
    });
    it("renders Footer", () => {
        render(<Footer />);
        expect(screen.getByRole("contentinfo")).toBeInTheDocument();
    });
    it("renders Footer", () => {
        render(<Footer />);
        expect(screen.getByRole("list")).toBeInTheDocument();
    });
});
