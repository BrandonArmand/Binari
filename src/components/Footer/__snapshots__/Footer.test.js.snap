// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders snapshot 1`] = `
<div>
  <footer
    class="makeStyles-footer-4"
  >
    <div
      class="makeStyles-container-7"
    >
      <div
        class="makeStyles-left-2"
      >
        <ul
          class="MuiList-root makeStyles-list-8 MuiList-padding"
        >
          <li
            class="MuiListItem-root makeStyles-inlineBlock-9 MuiListItem-gutters"
          >
            <a
              href="https://www.netlify.com"
            >
              <img
                alt="Deploys by Netlify"
                src="netlify.svg"
              />
            </a>
          </li>
          <li
            class="MuiListItem-root makeStyles-inlineBlock-9 MuiListItem-gutters"
          >
            <a
              class="makeStyles-block-1"
              href="http://brandonarmand.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Brandon A. Welsh
            </a>
          </li>
          <li
            class="MuiListItem-root makeStyles-inlineBlock-9 MuiListItem-gutters"
          >
            <a
              class="makeStyles-block-1"
              href="https://github.com/BrandonArmand/Binari/blob/master/LICENSE.md"
              rel="noopener noreferrer"
              target="_blank"
            >
              MIT License
            </a>
          </li>
        </ul>
      </div>
    </div>
  </footer>
</div>
`;
